### ISSUE TITLE
[Bug/Mod Incompatibility]: 

### ISSUE THREAD

**Issue by LinkachuGaming** [Soaryn/XyCraftTracker#61]
### Mod Title

XyCraft Machines

### Minecraft Version

1.21.0

### Severity

Crashing

### NeoForge Version

21.0.148

### Describe the Suggestion

During mod loading XYCraft seems to attempt to verify components if a mod has edited the components through ModifyDefaultComponents, (tv.soaryn.xycraft.machines.content.items.XynergyStorageItem.verifyComponentsAfterLoad(XynergyStorageItem.java:34))

However, this code attempts to read a ModConfig, which doesn't make sense in this environment as the config files have not been loaded yet.

### Reproduction Steps

1. Load up a NeoForge 21.0.148 environment, with both Your mod and my Mod, (Stackable127).
2. Launch the game
3. After a few moments, the game will crash with the given error: Cannot get config value before config is loaded.
4. Volia

### Additional Context

---- Minecraft Crash Report ----
// Don't do that.

Time: 2024-08-02 22:40:00
Description: Mod loading error has occurred

java.lang.Exception: Mod Loading has failed
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.logging.CrashReportExtender.dumpModLoadingCrashReport(CrashReportExtender.java:57) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.client.loading.ClientModLoader.completeModLoading(ClientModLoader.java:123) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.buildInitialScreens(Minecraft.java:679) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.onGameLoadFinished(Minecraft.java:652) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.onResourceLoadFinished(Minecraft.java:647) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.lambda$new$7(Minecraft.java:635) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.Util.ifElse(Util.java:516) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.lambda$new$8(Minecraft.java:629) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.client.loading.NeoForgeLoadingOverlay.render(NeoForgeLoadingOverlay.java:145) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.renderer.GameRenderer.render(GameRenderer.java:1087) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.runTick(Minecraft.java:1201) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.run(Minecraft.java:813) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.main.Main.main(Main.java:230) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.NeoForgeClientUserdevLaunchHandler.runService(NeoForgeClientUserdevLaunchHandler.java:23) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.4.jar%23125!/:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?] {}

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at MC-BOOTSTRAP/com.google.common@32.1.2-jre/com.google.common.base.Preconditions.checkState(Preconditions.java:512) ~[guava-32.1.2-jre.jar%23110!/:?] {}
-- MOD stackable --
Details:
	Caused by 0: java.lang.IllegalStateException: java.lang.IllegalStateException: Cannot get config value before config is loaded.
		at TRANSFORMER/stackable@1.2.2/com.lonkachu.stackable.Stackable.ModifyDefaultComponentsEvent(Stackable.java:118) ~[%23199!/:?] {re:mixin,re:classloading}
		at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.ConsumerEventHandler.invoke(ConsumerEventHandler.java:26) ~[bus-8.0.1.jar%23147!/:?] {}
		at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:350) ~[bus-8.0.1.jar%23147!/:?] {}
		at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:329) ~[bus-8.0.1.jar%23147!/:?] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModContainer.acceptEvent(ModContainer.java:177) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModLoader.lambda$postEvent$13(ModLoader.java:358) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModList.forEachModInOrder(ModList.java:164) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModLoader.postEvent(ModLoader.java:358) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.internal.RegistrationEvents.modifyComponents(RegistrationEvents.java:28) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
		at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.internal.RegistrationEvents.init(RegistrationEvents.java:21) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
		at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1804) ~[?:?] {}
		at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java) ~[?:?] {}
		at TRANSFORMER/minecraft@1.21/net.minecraft.server.packs.resources.SimpleReloadInstance.lambda$new$3(SimpleReloadInstance.java:86) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading}
		at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:151) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
		at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading}
		at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
		at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:111) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
		at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.runTick(Minecraft.java:1161) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.run(Minecraft.java:813) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at TRANSFORMER/minecraft@1.21/net.minecraft.client.main.Main.main(Main.java:230) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
		at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
		at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.NeoForgeClientUserdevLaunchHandler.runService(NeoForgeClientUserdevLaunchHandler.java:23) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.21.jar%23143!/:4.0] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.4.jar%23125!/:?] {}
		at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.4.jar%23125!/:?] {}
		at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?] {}
		at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?] {}

	Mod File: /C:/Users/Linka/Desktop/StackableForge/build/resources/main/
	Failure message: fml.modloadingissue.errorduringevent
	Failure message arg 1: net.neoforged.neoforge.event.ModifyDefaultComponentsEvent
	Mod Version: 1.2.2
	Mod Issue URL: NOT PROVIDED
	Exception message: java.lang.IllegalStateException: Cannot get config value before config is loaded.
Stacktrace:
	at MC-BOOTSTRAP/com.google.common@32.1.2-jre/com.google.common.base.Preconditions.checkState(Preconditions.java:512) ~[guava-32.1.2-jre.jar%23110!/:?] {}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.common.ModConfigSpec$ConfigValue.getRaw(ModConfigSpec.java:1228) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.common.ModConfigSpec$ConfigValue.get(ModConfigSpec.java:1215) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.common.ModConfigSpec$IntValue.getAsInt(ModConfigSpec.java:1308) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/xycraft_machines@0.7.33/tv.soaryn.xycraft.machines.content.items.XynergyStorageItem.verifyComponentsAfterLoad(XynergyStorageItem.java:34) ~[xycraft_machines-0.7.33.jar%23196!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.world.item.ItemStack.<init>(ItemStack.java:268) ~[neoforge-21.0.148.jar%23193!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:stackable.mixins.json:ItemStackFixin from mod stackable,pl:mixin:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.world.item.ItemStack.<init>(ItemStack.java:261) ~[neoforge-21.0.148.jar%23193!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:stackable.mixins.json:ItemStackFixin from mod stackable,pl:mixin:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.world.item.ItemStack.<init>(ItemStack.java:245) ~[neoforge-21.0.148.jar%23193!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:stackable.mixins.json:ItemStackFixin from mod stackable,pl:mixin:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.world.item.Item.getDefaultInstance(Item.java:401) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B}
	at TRANSFORMER/stackable@1.2.2/com.lonkachu.stackable.Stackable.lambda$ModifyDefaultComponentsEvent$0(Stackable.java:113) ~[%23199!/:?] {re:mixin,re:classloading}
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?] {}
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?] {}
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[?:?] {}
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?] {}
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?] {}
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:?] {}
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:?] {}
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?] {}
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?] {}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.event.ModifyDefaultComponentsEvent.modifyMatching(ModifyDefaultComponentsEvent.java:75) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/stackable@1.2.2/com.lonkachu.stackable.Stackable.ModifyDefaultComponentsEvent(Stackable.java:112) ~[%23199!/:?] {re:mixin,re:classloading}
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.ConsumerEventHandler.invoke(ConsumerEventHandler.java:26) ~[bus-8.0.1.jar%23147!/:?] {}
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:350) ~[bus-8.0.1.jar%23147!/:?] {}
	at MC-BOOTSTRAP/net.neoforged.bus/net.neoforged.bus.EventBus.post(EventBus.java:329) ~[bus-8.0.1.jar%23147!/:?] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModContainer.acceptEvent(ModContainer.java:177) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModLoader.lambda$postEvent$13(ModLoader.java:358) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModList.forEachModInOrder(ModList.java:164) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.ModLoader.postEvent(ModLoader.java:358) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.internal.RegistrationEvents.modifyComponents(RegistrationEvents.java:28) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at TRANSFORMER/neoforge@21.0.148/net.neoforged.neoforge.internal.RegistrationEvents.init(RegistrationEvents.java:21) ~[neoforge-21.0.148.jar%23194!/:?] {re:classloading}
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1804) ~[?:?] {}
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java) ~[?:?] {}
	at TRANSFORMER/minecraft@1.21/net.minecraft.server.packs.resources.SimpleReloadInstance.lambda$new$3(SimpleReloadInstance.java:86) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:151) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.21/net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:111) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.runTick(Minecraft.java:1161) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.Minecraft.run(Minecraft.java:813) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.21/net.minecraft.client.main.Main.main(Main.java:230) ~[neoforge-21.0.148.jar%23193!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?] {}
	at java.base/java.lang.reflect.Method.invoke(Method.java:580) ~[?:?] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:136) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:124) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.NeoForgeClientUserdevLaunchHandler.runService(NeoForgeClientUserdevLaunchHandler.java:23) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/fml_loader@4.0.21/net.neoforged.fml.loading.targets.CommonLaunchHandler.lambda$launchService$4(CommonLaunchHandler.java:118) ~[loader-4.0.21.jar%23143!/:4.0] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.run(Launcher.java:103) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.Launcher.main(Launcher.java:74) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-11.0.4.jar%23125!/:?] {}
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-11.0.4.jar%23125!/:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.run(BootstrapLauncher.java:210) [bootstraplauncher-2.0.2.jar:?] {}
	at cpw.mods.bootstraplauncher@2.0.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:69) [bootstraplauncher-2.0.2.jar:?] {}

-- System Details --
Details:
	Minecraft Version: 1.21
	Minecraft Version ID: 1.21
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 21.0.3, Amazon.com Inc.
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Amazon.com Inc.
	Memory: 189350712 bytes (180 MiB) / 734003200 bytes (700 MiB) up to 5343543296 bytes (5096 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 5600G with Radeon Graphics         
	Identifier: AuthenticAMD Family 25 Model 80 Stepping 0
	Microarchitecture: Zen 3
	Frequency (GHz): 3.89
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: AMD Radeon(TM) Graphics
	Graphics card #0 vendor: Advanced Micro Devices, Inc.
	Graphics card #0 VRAM (MiB): 4096.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 31.0.21902.5
	Graphics card #1 name: AMD Radeon RX 5700 XT
	Graphics card #1 vendor: Advanced Micro Devices, Inc.
	Graphics card #1 VRAM (MiB): 8176.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 31.0.22011.4008
	Graphics card #2 name: AMD Radeon RX 5700 XT
	Graphics card #2 vendor: Advanced Micro Devices, Inc.
	Graphics card #2 VRAM (MiB): 8176.00
	Graphics card #2 deviceId: VideoController3
	Graphics card #2 versionInfo: 31.0.24033.1003
	Graphics card #3 name: AMD Radeon(TM) Graphics
	Graphics card #3 vendor: Advanced Micro Devices, Inc.
	Graphics card #3 VRAM (MiB): 4096.00
	Graphics card #3 deviceId: VideoController4
	Graphics card #3 versionInfo: 31.0.21912.14
	Memory slot #0 capacity (MiB): 4096.00
	Memory slot #0 clockSpeed (GHz): 2.13
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MiB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.13
	Memory slot #1 type: DDR4
	Memory slot #2 capacity (MiB): 4096.00
	Memory slot #2 clockSpeed (GHz): 2.13
	Memory slot #2 type: DDR4
	Memory slot #3 capacity (MiB): 8192.00
	Memory slot #3 clockSpeed (GHz): 2.13
	Memory slot #3 type: DDR4
	Virtual memory max (MiB): 46152.24
	Virtual memory used (MiB): 39207.89
	Swap memory total (MiB): 25778.81
	Swap memory used (MiB): 2047.91
	Space in storage for jna.tmpdir (MiB): <path not set>
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): <path not set>
	Space in storage for io.netty.native.workdir (MiB): <path not set>
	Space in storage for java.io.tmpdir (MiB): available: 21699.93, total: 354963.00
	Space in storage for workdir (MiB): available: 21699.93, total: 354963.00
	JVM Flags: 1 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump
	ModLauncher: 11.0.4+main.d2e20e43
	ModLauncher launch target: forgeclientuserdev
	ModLauncher services: 
		sponge-mixin-0.14.0+mixin.0.8.6.jar mixin PLUGINSERVICE 
		loader-4.0.21.jar slf4jfixer PLUGINSERVICE 
		loader-4.0.21.jar runtime_enum_extender PLUGINSERVICE 
		at-modlauncher-10.0.1.jar accesstransformer PLUGINSERVICE 
		loader-4.0.21.jar runtimedistcleaner PLUGINSERVICE 
		modlauncher-11.0.4.jar mixin TRANSFORMATIONSERVICE 
		modlauncher-11.0.4.jar fml TRANSFORMATIONSERVICE 
	FML Language Providers: 
		javafml@4.0
		lowcodefml@4.0
		minecraft@4.0
	Mod List: 
		neoforge-21.0.148.jar                             |Minecraft                     |minecraft                     |1.21                |Manifest: a1:d4:5e:04:4f:d3:d6:e0:7b:37:97:cf:77:b0:de:ad:4a:47:ce:8c:96:49:5f:0a:cf:8c:ae:b2:6d:4b:8a:3f
		main                                              |Stackable                     |stackable                     |1.2.2               |Manifest: NOSIGNATURE
		xycraft_core-0.7.33-all.jar                       |XyCraft Core                  |xycraft_core                  |0.7.33              |Manifest: NOSIGNATURE
		neoforge-21.0.148.jar                             |NeoForge                      |neoforge                      |21.0.148            |Manifest: NOSIGNATURE
		xycraft_world-0.7.33.jar                          |XyCraft World                 |xycraft_world                 |0.7.33              |Manifest: NOSIGNATURE
		xycraft_override-0.7.33.jar                       |XyCraft Override              |xycraft_override              |0.7.33              |Manifest: NOSIGNATURE
		xycraft_machines-0.7.33.jar                       |XyCraft Machines              |xycraft_machines              |0.7.33              |Manifest: NOSIGNATURE
	Crash Report UUID: c3e37dcb-f6f3-4ee2-bb18-6b9f7beebe1c
	FML: 4.0.21
	NeoForge: 21.0.148Disconnected from the target VM, address: '127.0.0.1:59767', transport: 'socket'
	
https://github.com/LinkachuGaming/StackableForge/tree/1.21
https://modrinth.com/mod/stackable127/version/F98N0Rg2

**Comment by Soaryn** [comment]
Interesting, the event is called WAY early compared to everything else that modifies the components. I have added a check to the verify (given we have no control over the fact it is going to force a verify on modifying) to ensure the config is at least loaded. That method is the only method present that allows setting a component value by the time the level is loaded. And has worked thus far on server fine. May look at not doing it at all via components, but this way was allowing some mutations to be technically done via components in the future.

Fixed in next public version
